### PR TITLE
Update docs and organize code related to them

### PR DESCRIPTION
**Pull request by fgimenez**
This is the first PR of a set to add new documentation to project-infra, it: 

* Updates existing top-level readme and `github/ci/prow-deploy` readme.
* Removes unused kubernetes_crud ansible role and related presubmit.
* Moves all the ansible roles related to phx-prow to `github/ci/phx-prow`
* Removes dead code under `github/ci/prow-deploy`

/cc @dhiller @enp0s3 

Signed-off-by: Federico Gimenez <fgimenez@redhat.com>

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -2,16 +2,86 @@
...
+ Code to deploy CNI plugins (currently only sriov-passthrough-cni is available)
+
+ * external-plugins
+
+ Prow plugins used on our setup
+
  * github/ci
 
-  Ansible based prow provisioner for various CI/CD tasks in kubevirt.
+ Infrastructure code for ou main deployments:
```
Comment (dhiller) [inline README.md:15]:
```suggestion
 Infrastructure code for our main deployments:
```

File: github/ci/prow-deploy/README.md [hunk 2]
```diff
@@ -1,27 +1,34 @@
...
+This directory contains code to test and deploy Prow and related components
+in our clusters.
 
-- kubevirtci-testing
+## Continuous Delivery of Prow
 
-Is the testing environment to test prow both locally or on automatic jobs.
-It uses a cluster created by kubvirtci as target to deploy prow
+There are three Prow jobs defined in `kubevirt/project-infra` that help us to
+implement Continuous Delivery of the componenets we use in our Prow setup:
```
Comment (dhiller) [inline github/ci/prow-deploy/README.md:9]:
```suggestion
implement Continuous Delivery of the components we use in our Prow setup:
```

File: github/ci/prow-deploy/README.md [hunk 3]
```diff
@@ -1,27 +1,34 @@
...
 
-- ibmcloud-production
+* `periodic-project-infra-prow-bump`: periodic that gets the latest Prow manifests
+from `kubernetes/test-infra`, bumps the tags of the Prow images used in our
+setup and proposes a PR with the changes.
 
-Is the production environment. This is the default environment for production
-deployment. A deployment on this environment will affect real jobs
+Besides additional changes proposed by collaborators, the normal workflow for
+updating Prow consist of `periodic-project-infra-prow-bump` proposing a PR with
```
Comment (dhiller) [inline github/ci/prow-deploy/README.md:25]:
```suggestion
updating Prow consists of `periodic-project-infra-prow-bump` proposing a PR with
```

**Review by dhiller — COMMENTED** [review]
/approve

Looks very good in general, some nits I found. Thanks for doing this, very helpful!

**Comment by fgimenez** [comment]
@dhiller thanks! Addressed your suggestions and updated the main readme, PTAL.

**Review by dhiller — APPROVED** [review]
/approve

**Pull request merged**